internal/controllers/vgmanager/controller.go

```diff
@@ -105,6 +108,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	logger := log.FromContext(ctx)
 	logger.V(1).Info("reconciling")
 
+	// remove old entries because symlink can change
+	defer r.SymlinkResolver.Cleanup()
```

Comment:
I think we could just initialize a new map and let gc do the rest. then you dont need a cleanup or a close routine actually (you can completely drop that code and compiled behavior is the same in gc because you assign to zero value in Cleanup())